### PR TITLE
Amp page level reload

### LINKED ISSUE #6569: AMP - Page-level reload
An AMP page is currently refreshed in response to any change in the application. We should limit this to only refresh when the page itself has changed.

### PR DESCRIPTION
**Pull request by ijjk**
This makes sure an AMP page is only reloaded when it is changed instead of when any page in the project is changed. 

Closes: #6569

### CODE REVIEW

**Comment by timneutkens** [comment]
If @Timer ✅s you can merge

File: examples/with-amp/README.md [hunk 1]
```diff
@@ -1,26 +1,30 @@
-[![Deploy to now](https://deploy.now.sh/static/button.svg)](https://deploy.now.sh/?repo=https://github.com/zeit/next.js/tree/master/examples/with-amp)
+[![Deploy to now](https://deploy.now.sh/static/button.svg)](https://deploy.now.sh/?repo=https://github.com/zeit/next.js/tree/master/examples/experimental-amp)
```
Comment (Timer) [inline examples/with-amp/README.md]: This readme is inaccurate. Can we switch back to the old one with some updates?

File: examples/with-amp/package.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "with-amp",
+  "name": "experimental-amp",
```
Comment (Timer) [inline examples/with-amp/package.json]:
```suggestion
  "name": "with-amp",
```

File: packages/next/client/amp-dev.js [hunk 3]
```diff
@@ -10,6 +11,45 @@ if (!window.EventSource) {
...
+  /* eslint-disable-next-line */
+  return mostRecentHash !== __webpack_hash__
+}
+
+// Webpack disallows updates in other states.
+function canApplyUpdates () {
+  return module.hot.status() === 'idle'
+}
+
+// Attempt to update code on the fly, fall back to a hard reload.
```
Comment (Timer) [inline packages/next/client/amp-dev.js]:
```suggestion
// This function reads code updates on the fly and hard
// reloads the page when it has changed.
```

File: packages/next/client/amp-dev.js [hunk 4]
```diff
@@ -10,6 +11,45 @@ if (!window.EventSource) {
...
+  }
+  try {
+    const res = await fetch(`${hotUpdatePath}${curHash}.hot-update.json`)
+    const data = await res.json()
+    const curPage = page === '/' ? 'index' : page
+    const pageUpdated = Object.keys(data.c)
+      .some(mod => mod.indexOf(`pages/${curPage}`) !== -1)
+
+    if (pageUpdated) {
+      window.location.reload()
```
Comment (Timer) [inline packages/next/client/amp-dev.js]:
```suggestion
      document.location.reload(true)
```

File: packages/next/client/amp-dev.js [hunk 5]
```diff
@@ -10,6 +11,45 @@ if (!window.EventSource) {
...
+    } else {
+      curHash = mostRecentHash
+    }
+  } catch (err) {
+    console.error('Error occurred checking for update', err)
```
Comment (Timer) [inline packages/next/client/amp-dev.js]: This should probably fall back to reloading the page, right?

```suggestion
    console.error('Error occurred checking for update', err)
    document.location.reload(true)
```

**Review by Timer — CHANGES_REQUESTED** [review]

File: examples/with-amp/README.md (hunk 1 above)
Comment (ijjk) [reply examples/with-amp/README.md]: Yeah, updated

**Review by Timer — APPROVED** [review]

**Pull request merged**